### PR TITLE
Bug token lastused

### PR DESCRIPTION
**Pull request by slevine12**
Resolves: https://github.com/datumforge/datum/issues/1059

The `LastUsedAt` field is now updated when PAT and API tokens are used. The function `func updateLastUsed(ctx context.Context, dbClient *generated.Client, au *auth.AuthenticatedUser, tokenID string)` was added in the `auth.go` file to successfully update the `LastUsedAt` field. This function was referenced in `func Authenticate(conf *AuthOptions) echo.MiddlewareFunc` (also found in the `auth.go` file) in order to update this field while tokens are being authenticated. 

Tests were also implemented in both the `personalaccesstoken_test.go` and `apitoken_test.go` files to make sure `LastUsedAt` was being updated correctly.

### CODE REVIEW

File: internal/graphapi/apitoken_test.go [hunk 1]
```diff
@@ -368,3 +370,38 @@ func (suite *GraphTestSuite) TestMutationDeleteAPIToken() {
 		})
 	}
 }
+
+func (suite *GraphTestSuite) TestLastUsedAPIToken() {
+
```
Comment (golanglemonade) [inline internal/graphapi/apitoken_test.go:375]:
```suggestion
```

File: internal/graphapi/apitoken_test.go [hunk 2]
```diff
@@ -368,3 +370,38 @@ func (suite *GraphTestSuite) TestMutationDeleteAPIToken() {
...
+	t := suite.T()
+
+	defer mock_fga.ClearMocks(suite.client.fga)
+
+	// setup user context
+	reqCtx, err := userContext()
+	require.NoError(t, err)
+
+	// create new API token
+	token := (&APITokenTokenBuilder{client: suite.client}).MustNew(reqCtx, t)
```
Comment (golanglemonade) [inline internal/graphapi/apitoken_test.go:385]: This naming was fixed between your changes, so you need to update it to match.
```suggestion
	token := (&APITokenBuilder{client: suite.client}).MustNew(reqCtx, t)
```

File: internal/graphapi/apitoken_test.go [hunk 3]
```diff
@@ -368,3 +370,38 @@ func (suite *GraphTestSuite) TestMutationDeleteAPIToken() {
...
+
+	mock_fga.CheckAny(t, suite.client.fga, true)
+
+	// check that the last used is empty
+	res, err := suite.client.datum.GetAPITokenByID(reqCtx, token.ID)
+	require.NoError(t, err)
+	assert.Empty(t, res.APIToken.LastUsedAt)
+
+	// TODO: (slevine: update once we have updated the last used at field on the token when used)
+	// // setup graph client using the API token
```
Comment (golanglemonade) [inline internal/graphapi/apitoken_test.go:395]: remove old comment
```suggestion
	// setup graph client using the API token
```

File: internal/graphapi/personalaccesstoken_test.go [hunk 4]
```diff
@@ -426,15 +428,15 @@ func (suite *GraphTestSuite) TestLastUsedPersonalAccessToken() {
 
 	// TODO: (slevine: update once we have updated the last used at field on the token when used)
 	// // setup graph client using the personal access token
```
Comment (golanglemonade) [inline internal/graphapi/personalaccesstoken_test.go:430]:
```suggestion
	// setup graph client using the personal access token
```

File: pkg/middleware/auth/auth.go [hunk 5]
```diff
@@ -65,14 +65,16 @@ func Authenticate(conf *AuthOptions) echo.MiddlewareFunc {
 
 			var au *auth.AuthenticatedUser
 
+			var id string
+
```
Comment (golanglemonade) [inline pkg/middleware/auth/auth.go:69]: this is a nit, but if you are declaring multiple variables in a row its usually better to do: 

```suggestion
			var  (
				au *auth.AuthenticatedUser
				id string
			)
```

**Review by golanglemonade — APPROVED** [review]

**Review by matoszz — APPROVED** [review]

**Pull request merged**